src/statement.rs

```diff
@@ -350,10 +359,47 @@ pub(crate) fn parse_statement<'a>(
         Token::Ident(_, Keyword::CASE) => Some(Statement::Case(parse_case_statement(parser)?)),
         Token::Ident(_, Keyword::COPY) => Some(Statement::Copy(parse_copy_statement(parser)?)),
         Token::Ident(_, Keyword::DO) => Some(parse_do(parser)?),
+        Token::Ident(_, Keyword::TRUNCATE) => Some(parse_truncate_table(parser)?),
         _ => None,
     })
 }
 
+pub(crate) fn parse_truncate_table<'a>(
+    parser: &mut Parser<'a, '_>,
+) -> Result<Statement<'a>, ParseError> {
+    let truncate_span = parser.consume_keyword(Keyword::TRUNCATE)?;
+    let mut table_span = None;
+    let table_name = loop {
+        match parser.token {
+            Token::Ident(_, k) => {
+                if k == Keyword::TABLE {
+                    if None == table_span {
+                        table_span = Some(parser.span.clone());
+                    } else {
+                        parser.expected_failure(parser.token.name())?;
+                    };
+                } else if k == Keyword::QUOTED_IDENTIFIER || k == Keyword::NOT_A_KEYWORD {
+                    break parse_qualified_name(parser)?;
+                } else {
+                    parser.expected_failure(parser.token.name())?;
+                }
+            }
+            _ => parser.expected_failure(parser.token.name())?,
+        };
+        parser.next();
+    };
+    parser.next();
+    match parser.token {
+        Token::Eof => {}
+        _ => parser.expected_failure("illegal Eof!")?,
+    }
```

Comment:
I belive this part could (and should be written as), though I have not tested it
```rust
let table_span = parser.skip_keyword(Keyword::TABLE);
let table_name = parse_qualified_name(parser)?; 
```
We should not check for EOF here, since the truncate_table statement may be part of a bigger list of statements seperated by ; or be inside some kind of begin/end block